File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/PaymentProduct5500SpecificOutputMapper.cs
using PaymentProduct5500SpecificOutputSdk = OnlinePayments.Sdk.Domain.PaymentProduct5500SpecificOutput;
using PaymentProduct5500SpecificOutputDto = Business.Domain.Payments.PaymentDetails.PaymentProduct5500SpecificOutput;

namespace Infrastructure.Mappers.PaymentDetails;

public static class PaymentProduct5500SpecificOutputMapper
{
    public static PaymentProduct5500SpecificOutputDto Map(PaymentProduct5500SpecificOutputSdk? response)
    {
        return new()
        {
            PaymentReference = response?.PaymentReference,
            PaymentEndDate = response?.PaymentEndDate,
            PaymentStartDate = response?.PaymentStartDate,
            EntityId = response?.EntityId
        };
    }
}

File: server/nodejs/src/infrastructure/mappers/payment-details/payment-product-5500-specific-output-mapper.ts
import type { Domain } from 'onlinepayments-sdk-nodejs';
import { PaymentProduct5500SpecificOutput } from '../../../business/domain/payments/payment-details/payment-product-5500-specific-output';

type PaymentProduct5500SpecificOutputSdk = Domain.PaymentProduct5500SpecificOutput;

export const PaymentProduct5500SpecificOutputMapper = {
    fromSdkResponse: (
        response?: PaymentProduct5500SpecificOutputSdk | null,
    ): PaymentProduct5500SpecificOutput | undefined => {
        if (!response) {
            return undefined;
        }

        return {
            paymentReference: response?.paymentReference,
            paymentEndDate: response?.paymentEndDate,
            paymentStartDate: response?.paymentStartDate,
            entityId: response?.entityId,
        };
    },
};
